frontend/src/views/components/Legend/LegendTimeline/LegendTimeline.component.tsx

```diff
@@ -0,0 +1,43 @@
+import Datepicker from 'views/shared/datepicker/component';
+import { useCallback } from 'react';
+import type { Timeline } from 'types/layer';
+
+interface LegendTimelineProps {
+  timeline: Timeline;
+  layerName: string;
+  setDate: (date: Date) => void;
+  selected: Date;
+}
+
+const LegendTimeline = ({ timeline, layerName, setDate, selected }: LegendTimelineProps) => {
+  const { startDate, endDate, period, steps } = timeline;
+
+  const getResolution = useCallback(() => {
+    const resolutionPeriods = {
+      yearly: 'year',
+      monthly: 'month',
+      daily: 'day',
+    };
+    return resolutionPeriods[period];
+  }, [period]);
+
+  return (
+    <>
+      <div className="m-legend-timeline">
+        <span className="timeline-title">Display {layerName} in</span>
```

Comment:
This should be a label with the `htmlFor` attribute set to the same value as an `id` attribute that you should add on the `Datepicker` component.